lxd/instance/instance_utils.go

```diff
@@ -125,8 +125,43 @@ func exclusiveConfigKeys(key1 string, key2 string, config map[string]string) (va
 	return "", false, nil
 }
 
+var containerSpecificConfigKeys = []string{
```

Comment:
I think I would have preferred using a `map[instancetype.Type][]string` and then stored both container and VM specific keys in the same map, that way they can be looked up by the provided `instanceType` argument in `ValidConfig`.